include/exadg/fluid_structure_interaction/driver_precice_fluid.h

```diff
@@ -0,0 +1,594 @@
+      this->timer_tree.insert({"FSI", "Setup", "ALE"}, timer_local.wall_time());
+    }
+    /******************************************* ALE ********************************************/
+
+    /****************************************** FLUID *******************************************/
+    {
+      dealii::Timer timer_local;
+      timer_local.restart();
+
+      // initialize fluid_operator
+      fluid_operator =
+        IncNS::create_operator<dim, Number>(this->application->get_grid_fluid(),
+                                            fluid_grid_motion,
+                                            this->application->get_boundary_descriptor_fluid(),
+                                            this->application->get_field_functions_fluid(),
+                                            this->application->get_parameters_fluid(),
+                                            "fluid",
+                                            this->mpi_comm);
+
+      // initialize matrix_free
+      fluid_matrix_free_data = std::make_shared<MatrixFreeData<dim, Number>>();
+      fluid_matrix_free_data->append(fluid_operator);
+
+      fluid_matrix_free = std::make_shared<dealii::MatrixFree<dim, Number>>();
+      if(this->application->get_parameters_fluid().use_cell_based_face_loops)
+        Categorization::do_cell_based_loops(*this->application->get_grid_fluid()->triangulation,
+                                            fluid_matrix_free_data->data);
+      std::shared_ptr<dealii::Mapping<dim> const> mapping =
+        get_dynamic_mapping<dim, Number>(this->application->get_grid_fluid(), fluid_grid_motion);
+      fluid_matrix_free->reinit(*mapping,
+                                fluid_matrix_free_data->get_dof_handler_vector(),
+                                fluid_matrix_free_data->get_constraint_vector(),
+                                fluid_matrix_free_data->get_quadrature_vector(),
+                                fluid_matrix_free_data->data);
+
+      // setup Navier-Stokes operator
+      fluid_operator->setup(fluid_matrix_free, fluid_matrix_free_data);
+
+      // setup postprocessor
+      fluid_postprocessor = this->application->create_postprocessor_fluid();
+      fluid_postprocessor->setup(*fluid_operator);
+
+      this->timer_tree.insert({"FSI", "Setup", "Fluid"}, timer_local.wall_time());
+    }
+    /****************************************** FLUID *******************************************/
+
+
+    /*********************************** INTERFACE COUPLING *************************************/
+
+    // writing
+    this->precice =
+      std::make_shared<ExaDG::preCICE::Adapter<dim, dim, VectorType>>(this->precice_parameters,
+                                                                      this->mpi_comm);
+
+    this->precice->add_write_surface(this->application->get_boundary_descriptor_fluid()
+                                       ->velocity->dirichlet_cached_bc.begin()
+                                       ->first,
```

Comment:
Does this mean that precice is restricted to a single boundary ID at the FSI interface at the moment? If so, we should highlight this in the code whereever this occurs, since the ExaDG-FSI variant is different in this regard.

Reply:
The current implementation: yes. I never felt the necessity to go beyond because one can always 'summarize' the related boundary IDs into a single one related to the coupling. I would also put it an extension on the TODO list if that's ok?

Reply:
Yes, it would be good to have the generality of other modules of ExaDG (with several different boundary IDs), so that the codes looks more "uniform".

Reply:
Added a todo comment